utils.sh

```diff
@@ -0,0 +1,242 @@
+}
+
+function runDapTests() {
+  echo '-----------------------test.sh------------------------------'
+  echo "Running DAP tests"
+  echo '------------------------------------------------------------'
+
+  dap_test_cid=$(docker-compose ps -q test-dap)
+
+  api_key_admin=$(docker-compose exec -T client conjur user rotate_api_key)
+
+  # Execute DAP tests
+  docker exec \
+  -e CONJUR_AUTHN_API_KEY=${api_key_admin} \
+  -e CONJUR_AUTHN_LOGIN="admin" \
+  ${dap_test_cid} \
+    mvn test
+}
+
+function createOssTestEnvironment() {
+  echo '-----------------------test.sh------------------------------'
+  echo "Creating OSS test environment"
+  echo '------------------------------------------------------------'
+
+  # Build OSS test container & start the cluster
+  docker-compose build --pull client conjur postgres test test-https conjur-proxy-nginx
+  export CONJUR_APPLIANCE_URL="http://conjur:3000"
+  docker-compose up -d client conjur postgres test-https
+
+  # Delay to allow time for conjur to come up
+  # TODO: remove this once we have HEALTHCHECK in place
+  echo 'Waiting for conjur server to be healthy'
+  docker-compose run --rm test ./wait_for_server.sh
+}
+
+function loadOssTestPolicy() {
+  echo '-----------------------test.sh------------------------------'
+  echo "Loading OSS test policy"
+  echo '------------------------------------------------------------'
+
+  conjur_client_cid=$(docker-compose ps -q client)
+
+  api_key=$(docker-compose exec -T conjur rails r "print Credentials['cucumber:user:admin'].api_key")
+
+  # Copy test-policy into a /tmp/test-policy within the possum container
+  docker cp test-policy ${conjur_client_cid}:/tmp
+
+  docker exec -e CONJUR_AUTHN_API_KEY=${api_key} \
+    ${conjur_client_cid} \
+    conjur policy load root /tmp/test-policy/root.yml
+}
+
+function printOssProxyConfiguration() {
+  echo '-----------------------test.sh------------------------------'
+  echo "Print Nginx proxy server configuration"
+  echo '------------------------------------------------------------'
+  echo 'Note: in order to change configuration, you need to edit the file: default.conf '
+
+  conjur_proxy_cid=$(docker-compose ps -q conjur-proxy-nginx)
+
+  exec_command='nginx-debug -T'
+  docker exec ${conjur_proxy_cid} ${exec_command}
+}
+
+function initializeOssCert() {
+  echo '-----------------------test.sh------------------------------'
+  echo "Fetch certificate for OSS using client cli"
+  echo '------------------------------------------------------------'
+
+  conjur_client_cid=$(docker-compose ps -q client)
+
+  # Get the pem file from conjur server
+  CONJUR_ACCOUNT="cucumber"
+  CONJUR_PROXY="https://conjur-proxy-nginx"
+
+  echo "remove old pem file"
+  rm -rf /test-cert/*
+
+  echo "fetch pem file from proxy https server"
+  exec_command='echo yes | conjur init -u '${CONJUR_PROXY}' -a '${CONJUR_ACCOUNT}' > tmp.out 2>&1'
+  docker exec ${conjur_client_cid} /bin/bash -c "$exec_command"
```

Comment:
Quote all of these vars you pass to docker exec in this file. You may need to use bash arrays to do it but it's 10x more resilient than counting on bash to properly split the args

Reply:
I copied this code from `./test.sh` and placed it into a shared file.
I know it is not the best, but I think this is fine for now since I plan on re-writing the unit tests as a whole and the `./test.sh` and `./dev.sh` will change for sure.
The tests are currently being executed 3 times against 3 different environments, I would rather spin up 3 environments and then have 3 times the amount of unit tests. 

I am looking at this as a quick solution for now until the tests are in better shape.